packages/core/src/components/graphs/alluvial.ts

```diff
@@ -0,0 +1,596 @@
+				}
+				return '';
+			})
+			.attr('y', 20)
+			.attr('x', (d, i) => {
+				const { width } = this.parent
+					.select(`text#category-${i}`)
+					.node()
+					.getBBox();
+				// Make the text on the left on node group (except first column)
+				let x = 0;
+				if (d + x >= width) {
+					x = -width + 4;
+				}
+				return x;
+			});
+
+		// Draws the links (Waves)
+		svg.append('g')
+			.attr('fill', 'none')
+			.selectAll('g')
+			.data(this.graph.links)
+			.join('path')
+			.classed('link', true)
+			.attr('d', sankeyLinkHorizontal())
+			.attr('id', (d) => `line-${d.index}`)
+			.attr('class', (d) => {
+				// Use a single color for the lines
+				if (options.monochrome) {
+					return this.model.getColorClassName({
+						classNameTypes: [ColorClassNameTypes.STROKE],
+						dataGroupName: 0,
+						originalClassName: 'link',
+					});
+				}
+
+				return this.model.getColorClassName({
+					classNameTypes: [ColorClassNameTypes.STROKE],
+					dataGroupName: d.source.index,
+					originalClassName: 'link',
+				});
+			})
+			.attr('stroke-width', (d) => Math.max(1, d.width))
+			.attr('stroke-opacity', Configuration.alluvial.opacity.default)
+			.attr(
+				'aria-label',
+				(d) =>
+					`${d.source.name} → ${d.target.name} (${d.value}
+						${options.units ? ' ' + options.units : ''})`
+			);
+
+		// Creating the groups
+		const node = svg
+			.append('g')
+			.selectAll('g')
+			.data(this.graph.nodes)
+			.enter()
+			.append('g')
+			.attr('id', (d) => `node-${d.index}`)
+			.classed('node-group', true)
+			.attr('transform', (d) => `translate(${d.x0}, ${d.y0})`);
+
+		// Creating the nodes
+		node.append('rect')
+			.classed('node', true)
+			.attr('height', (d) => d.y1 - d.y0)
+			.attr('width', (d) => d.x1 - d.x0)
+			.attr('fill', 'black');
+
+		// Group to hold the text & rectangle background
+		const textNode = node
+			.append('g')
+			.attr('id', (d) => `node-title-${d.index}`);
+
+		// Node title - text
+		textNode
+			.append('text')
+			.attr('id', (d) => `node-text-${d.index}`)
+			.attr('class', 'node-text')
+			.style('font-size', '12px')
+			.attr('text-anchor', 'start')
+			.attr('fill', 'white')
+			// Padding to text
+			.attr('x', 4)
+			// shift 13 pixels down to fit background container
+			.attr('dy', 13)
+			.text((d) => {
+				return `${d.name} (${d.value})`;
+			})
+			.attr('aria-label', (d) => {
+				return `${d.name} (${d.value})`;
+			});
+
+		// Text background
+		textNode
+			.append('rect')
+			.classed('node-text-bg', true)
+			.attr('width', (d, i) => {
+				// Determine rectangle width based on text width
+				const { width } = this.parent
+					.select(`text#node-text-${i}`)
+					.node()
+					.getBBox();
+				return width + 8;
+			})
+			.attr('height', 18)
+			.attr('stroke-width', 2)
+			.lower();
+
+		// Position group based on text width
+		textNode.attr('transform', (d, i) => {
+			const { width } = this.parent
+				.select(`text#node-text-${i}`)
+				.node()
+				.getBBox();
```

Comment:
getBBox() is known to have issues in firefox, depending on the version I think & the conditions of the element. We should probably use

```
const { width } = DOMUtils.getSVGElementSize(..., {
	useBBox: true,
});
```